diff --git a/lisp/ob-core.el b/lisp/ob-core.el
index af2c9912e..f2b550fb8 100644
--- a/lisp/ob-core.el
+++ b/lisp/ob-core.el
@@ -711,7 +711,11 @@ block."
 		     (let ((d (file-name-as-directory (expand-file-name dir))))
 		       (make-directory d 'parents)
 		       d))))
-		 (cmd (intern (concat "org-babel-execute:" lang)))
+		 ;; Override with :servant to access the process we will call
+		 ;; e.g. (:servant . "calculator")
+		 (cmd (if (assoc :servant params)
+			#'org-babel-servant
+			(intern (concat "org-babel-execute:" lang))))
 		 result)
 	    (unless (fboundp cmd)
 	      (error "No org-babel-execute function for %s!" lang))
